service_thing_aggregate/src/main/scala/dao/ObjectSearchDao.scala

```diff
@@ -0,0 +1,204 @@
+package dao
+
+import com.google.inject.Inject
+import models.MusitThing
+import models.dto.MusitThingDto
+import no.uio.musit.service.MusitResults._
+import play.api.Logger
+import play.api.db.slick.{DatabaseConfigProvider, HasDatabaseConfigProvider}
+import play.api.libs.concurrent.Execution.Implicits.defaultContext
+import slick.driver.JdbcProfile
+
+import scala.concurrent.Future
+
+/**
+ * Created by jarle on 06.10.16.
+ */
+
+sealed trait FieldValue
+
+case class EmptyValue() extends FieldValue
+
+case class LiteralValue(v: String) extends FieldValue
+
+/*If v contains a value which needs to be escaped, escapeChar contains the appropriate escape character.
+If v doesn't contains a value which needs to be escaped, escapeChar =
+ */
+case class WildcardValue(v: String, escapeChar: Char) extends FieldValue
+
+//#If we (in the future) need to treat some characters as invalid: case class InvalidValue(errorMessage: String) extends FieldValue
+
+class ObjectSearchDao @Inject() (
+    val dbConfigProvider: DatabaseConfigProvider
+) extends HasDatabaseConfigProvider[JdbcProfile] {
+
+  val logger = Logger(classOf[ObjectAggregationDao])
+
+  import driver.api._
+
+  private val musitThingTable = TableQuery[MusitThingTable]
+
+  def testInsert(museumId: Int, mthing: MusitThing): Future[Long] = {
```

Comment:
Short answer is; no we can't add test related code into the production code base.

Instead, you can as mentioned in a comment above, test with a limit that is much lower than 100 to verify the behaviour.

Eventually though, we should have a better way of adding data to our unit and integration tests . But for now, it is sufficient to use inserts in this case. Since the service is read only, its not wrong to assume that data is inserted using the evolution scripts in test and it. If you need to clarify where the data comes from, you can add a properly formatted comment to the test class.